contentcuration/contentcuration/management/commands/exportchannel.py

```diff
@@ -245,76 +247,105 @@ def process_assessment_metadata(ccnode, kolibrinode):
-    elif assessment_item.type == exercises.SINGLE_SELECTION:
-        template = 'perseus/multiple_selection.json'
-    elif assessment_item.type == exercises.FREE_RESPONSE:
-        template = 'perseus/input_question.json'
-    elif assessment_item.type == exercises.INPUT_QUESTION:
-        template = 'perseus/input_question.json'
-    elif assessment_item.type == exercises.PERSEUS_QUESTION:
-        template = 'perseus/perseus_question.json'
-
-    result = render_to_string(template, context).encode('utf-8', "ignore")
-    filename = "{0}.json".format(assessment_item.assessment_id)
-    zf.writestr(filename, result)
+    if assessment_item.type != exercises.PERSEUS_QUESTION:
+        template=''
+        if assessment_item.type == exercises.MULTIPLE_SELECTION:
+            template = 'perseus/multiple_selection.json'
+        elif assessment_item.type == exercises.SINGLE_SELECTION:
+            template = 'perseus/multiple_selection.json'
+        elif assessment_item.type == exercises.INPUT_QUESTION:
+            template = 'perseus/input_question.json'
+        elif assessment_item.type == exercises.PERSEUS_QUESTION:
+            template = 'perseus/perseus_question.json'
+        else:
+            raise TypeError("Unrecognized question type on item {}".format(assessment_item.assessment_id))
+
+        question, question_images = process_image_strings(assessment_item.question)
+
+        answer_data = json.loads(assessment_item.answers)
+        for answer in answer_data:
+            answer['answer'] = answer['answer'].replace(exercises.CONTENT_STORAGE_PLACEHOLDER, PERSEUS_IMG_DIR)
+            # In case perseus doesn't support =wxh syntax, use below code
+            # answer['answer'], answer_images = process_image_strings(answer['answer'])
+            # answer.update({'images': answer_images})
+
+        hint_data = json.loads(assessment_item.hints)
+        for hint in hint_data:
+            hint['hint'], hint_images = process_image_strings(hint['hint'])
+            hint.update({'images': hint_images})
+
+        context = {
+            'question': question,
+            'question_images': question_images,
+            'answers': sorted(answer_data, lambda x,y: cmp(x.get('order'), y.get('order'))),
+            'multiple_select': assessment_item.type == exercises.MULTIPLE_SELECTION,
+            'raw_data': assessment_item.raw_data.replace(exercises.CONTENT_STORAGE_PLACEHOLDER, PERSEUS_IMG_DIR),
+            'hints': sorted(hint_data, lambda x,y: cmp(x.get('order'), y.get('order'))),
+            'randomize': assessment_item.randomize,
+        }
+
+        result = render_to_string(template, context).encode('utf-8', "ignore")
+        write_to_zipfile("{0}.json".format(assessment_item.assessment_id), result, zf)
+
+def process_image_strings(content):
+    image_list = []
+    content = content.replace(exercises.CONTENT_STORAGE_PLACEHOLDER, PERSEUS_IMG_DIR)
+    for match in re.finditer(ur'!\[(?:.*)]\((.+)\)', content):
```

Comment:
Can you move this regex into its own variable and add a comment as to what this matches?